sdk/solana/src/ntt.ts

```diff
@@ -0,0 +1,805 @@
+import BN from "bn.js";
+import type { NativeTokenTransfer } from "./anchor-idl/index.js";
+import { idl } from "./anchor-idl/index.js";
+import {
+  BPF_LOADER_UPGRADEABLE_PROGRAM_ID,
+  TransferArgs,
+  nttAddresses,
+  programDataAddress,
+} from "./utils.js";
+
+export type Config = IdlAccounts<NativeTokenTransfer>["config"];
+export type InboxItem = IdlAccounts<NativeTokenTransfer>["inboxItem"];
+
+export class SolanaNttWormholeTransceiver<
+  N extends Network,
+  C extends SolanaChains
+> implements NttTransceiver<N, C, WormholeNttTransceiver.VAA>
+{
+  constructor(readonly manager: SolanaNtt<N, C>, readonly address: string) {
+    //
+  }
+
+  async *receive(
+    attestation: WormholeNttTransceiver.VAA,
+    sender?: AccountAddress<C> | undefined
+  ): AsyncGenerator<UnsignedTransaction<N, C>, any, unknown> {
+    throw new Error("Method not implemented.");
+  }
+
+  async *setPeer(
+    peer: ChainAddress<C>
+  ): AsyncGenerator<UnsignedTransaction<N, C>, any, unknown> {
+    throw new Error("Method not implemented.");
+  }
+}
+
+export class SolanaNtt<N extends Network, C extends SolanaChains>
+  implements Ntt<N, C>
+{
+  core: SolanaWormholeCore<N, C>;
+  xcvrs: SolanaNttWormholeTransceiver<N, C>[];
+  program: Program<NativeTokenTransfer>;
+  pdas: ReturnType<typeof nttAddresses>;
+
+  config?: Config;
+
+  constructor(
+    readonly network: N,
+    readonly chain: C,
+    readonly connection: Connection,
+    readonly contracts: Contracts & { ntt?: Ntt.Contracts }
+  ) {
+    if (!contracts.ntt) throw new Error("Ntt contracts not found");
+
+    this.program = new Program<NativeTokenTransfer>(
+      // @ts-ignore
+      idl.ntt,
+      this.contracts.ntt!.manager,
+      { connection }
+    );
+
+    this.core = new SolanaWormholeCore<N, C>(
+      network,
+      chain,
+      connection,
+      contracts
+    );
+    this.pdas = nttAddresses(this.program.programId);
+    this.xcvrs = [
+      new SolanaNttWormholeTransceiver<N, C>(
+        this,
+        this.contracts.ntt!.transceiver.wormhole
+      ),
+    ];
+  }
+
+  static async fromRpc<N extends Network>(
+    provider: Connection,
+    config: ChainsConfig<N, SolanaPlatformType>
+  ): Promise<SolanaNtt<N, SolanaChains>> {
+    const [network, chain] = await SolanaPlatform.chainFromRpc(provider);
+    const conf = config[chain]!;
+
+    if (conf.network !== network)
+      throw new Error(`Network mismatch: ${conf.network} != ${network}`);
+    if (!conf.tokenMap) throw new Error("Token map not found");
+
+    return new SolanaNtt(network as N, chain, provider, {
+      ...conf.contracts,
+      ntt: {
+        token: "",
+        manager: "",
+        transceiver: { wormhole: "" },
+      },
+    });
+  }
+
+  async getConfig(): Promise<Config> {
+    this.config =
+      this.config ??
+      (await this.program.account.config.fetch(this.pdas.configAccount()));
+    return this.config;
+  }
+
+  async getTokenDecimals(): Promise<number> {
+    // TODO:
+    return 9;
```

Comment:
We can get the `mint` address from the `config` and then fetch the token metadata for the decimals https://github.com/wormhole-foundation/example-native-token-transfers/blob/1193a4a00a8b32aeb0f8705913f09f9bfc3e53c2/solana/programs/example-native-token-transfers/src/config.rs#L17

Reply:
added this